os/sys/log.c

```diff
@@ -84,35 +85,9 @@ struct log_module all_modules[] = {
 void
 log_6addr(const uip_ipaddr_t *ipaddr)
 {
-  uint16_t a;
-  unsigned int i;
-  int f;
-
-  if(ipaddr == NULL) {
-    LOG_OUTPUT("(NULL IP addr)");
-    return;
-  }
-
-  if(ip64_addr_is_ipv4_mapped_addr(ipaddr)) {
-    /* Printing IPv4-mapped addresses is done according to RFC 4291 */
-    LOG_OUTPUT("::FFFF:%u.%u.%u.%u", ipaddr->u8[12], ipaddr->u8[13], ipaddr->u8[14], ipaddr->u8[15]);
-  } else {
-    for(i = 0, f = 0; i < sizeof(uip_ipaddr_t); i += 2) {
-      a = (ipaddr->u8[i] << 8) + ipaddr->u8[i + 1];
-      if(a == 0 && f >= 0) {
-        if(f++ == 0) {
-          LOG_OUTPUT("::");
-        }
-      } else {
-        if(f > 0) {
-          f = -1;
-        } else if(i > 0) {
-          LOG_OUTPUT(":");
-        }
-        LOG_OUTPUT("%x", a);
-      }
-    }
-  }
+  char buf[40];
```

Comment:
Should we change 40 to some define that states max IPv6 address length in string format, specified in uiplib.h?